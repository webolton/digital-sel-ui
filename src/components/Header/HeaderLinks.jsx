import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import isNil from 'lodash/isNil';
import withStyles from '@material-ui/core/styles/withStyles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Tooltip from '@material-ui/core/Tooltip';
import CustomDropdown from 'components/CustomDropdown/CustomDropdown';
import Button from 'components/CustomButtons/Button';
import headerLinksStyle from 'assets/javascripts/components/headerLinksStyle';

function HeaderLinks({ ...props }) {
  const { classes, currentUser } = props;
  const LogInButton = (() => (
    <ListItem className={classes.listItem}>
      <Tooltip
        id="login-button"
        title="Login to save textual notes"
        placement={window.innerWidth > 959 ? 'top' : 'left'}
        classes={{ tooltip: classes.tooltip }}
      >
        <Button
          href="google.com"
          target="_blank"
          color="transparent"
          className={classes.navLink}
        >
          Login
        </Button>
      </Tooltip>
    </ListItem>
  ));
  return (
    <List className={classes.list}>
      { isNil(currentUser) ? <LogInButton /> : null }
      <ListItem className={classes.listItem}>
        <CustomDropdown
          noLiPadding
          buttonText="William Bolton"
          buttonProps={{
            className: classes.navLink,
            color: 'transparent',
          }}
          dropdownList={[
            <Link to="/" className={classes.dropdownLink}>
              User Profile
            </Link>,
            <Link to="/" className={classes.dropdownLink}>
              Logout
            </Link>,
          ]}
        />
      </ListItem>
    </List>
  );
}

HeaderLinks.propTypes = {
  classes: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  const { authentication } = state;
  const { currentUser } = authentication;
  return {
    currentUser
  };
}

export default connect(mapStateToProps)(withStyles(headerLinksStyle)(HeaderLinks));
