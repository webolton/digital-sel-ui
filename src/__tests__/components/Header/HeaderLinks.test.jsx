import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import HeaderLinks from 'components/Header/HeaderLinks';

const initialState = { authentication: '' };
const mockStore = configureStore();
const store = mockStore(initialState);

describe('<HeaderLinks />', () => {
  describe('render() when there is no currentUser', () => {
    test('renders the component', () => {
      const wrapper = shallow(<HeaderLinks store={store} />);
      const component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
