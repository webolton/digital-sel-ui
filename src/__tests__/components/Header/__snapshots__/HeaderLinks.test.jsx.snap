// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLinks /> render() when there is no currentUser renders the component 1`] = `
<HeaderLinks
  classes={
    Object {
      "dropdownLink": "HeaderLinks-dropdownLink-10",
      "icons": "HeaderLinks-icons-8",
      "list": "HeaderLinks-list-1",
      "listItem": "HeaderLinks-listItem-2",
      "listItemText": "HeaderLinks-listItemText-3",
      "marginRight5": "HeaderLinks-marginRight5-12",
      "navLink": "HeaderLinks-navLink-4",
      "navLinkActive": "HeaderLinks-navLinkActive-7",
      "notificationNavLink": "HeaderLinks-notificationNavLink-5",
      "registerNavLink": "HeaderLinks-registerNavLink-6",
      "socialIcons": "HeaderLinks-socialIcons-9",
      "tooltip": "HeaderLinks-tooltip-11",
    }
  }
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
